### PR TITLE
feat: check if has associated credetials before updating the credential type

### PR DESCRIPTION
**Pull request by hsong-rh**
https://issues.redhat.com/browse/AAP-22563

UI will show the following error if credentials exist.
image

### CODE REVIEW

**Comment by hsong-rh** [comment]
> @hsong-rh The message remains the same for inputs/injectors shouldn't it change based on the field name. Also I think the errors should be reported with the field name so the UI can display it properly. 

Updated

File: src/aap_eda/api/views/credential_type.py [hunk 1]
```diff
@@ -122,6 +122,22 @@ def partial_update(self, request, pk):
...
 
+        if models.EdaCredential.objects.filter(
+            credential_type=credential_type
+        ).exists():
+            inputs = request.data.get("inputs")
+            injectors = request.data.get("injectors")
+            if inputs or injectors:
+                field = "inputs" if inputs else "injectors"
+                error = (
+                    "Modifications to inputs are not allowed for credential "
```
Comment (mkanoor) [inline src/aap_eda/api/views/credential_type.py:133]: @hsong-rh should this be
```
f"Modifications to {field} are not allowed for credential "
```

**Review by mkanoor — APPROVED** [review]

**Comment by eclarizio** [comment]
Very small nitpick: shouldn't this be "feat" and not "chore"? "chore" (if I understand it correctly) is for something that is internal to the project like adjusting a variable name, refactoring a method, or more broadly upgrading versions of plugins, things like that. This, however, adds new functionality that can be seen directly in the UI and seems more in line with a feature

**Comment by hsong-rh** [comment]
> Very small nitpick: shouldn't this be "feat" and not "chore"? "chore" (if I understand it correctly) is for something that is internal to the project like adjusting a variable name, refactoring a method, or more broadly upgrading versions of plugins, things like that. This, however, adds new functionality that can be seen directly in the UI and seems more in line with a feature

@eclarizio Got you, I thought it meant minor changes.

**Review by lgalis — APPROVED** [review]

File: src/aap_eda/api/views/credential_type.py [hunk 2]
```diff
@@ -122,6 +122,22 @@ def partial_update(self, request, pk):
                 {"errors": error}, status=status.HTTP_400_BAD_REQUEST
             )
 
+        if models.EdaCredential.objects.filter(
+            credential_type=credential_type
+        ).exists():
+            inputs = request.data.get("inputs")
+            injectors = request.data.get("injectors")
+            if inputs or injectors:
+                field = "inputs" if inputs else "injectors"
```
Comment (bzwei) [inline src/aap_eda/api/views/credential_type.py:131]: What if both fields are changed? The message only says inputs.
Maybe we don't pick field here, simply says Modifications to inputs or injectors are not allowed

**Review by bzwei — APPROVED** [review]

**Pull request merged**